client/packages/system/src/Immunisation/api/hooks/useVaccineCourse.ts

```diff
@@ -137,18 +148,61 @@ const useGet = (id: string) => {
   return query;
 };
 
-const useCreate = () => {
+const useCreate = (setErrorMessage: Dispatch<SetStateAction<string>>) => {
   const { api, storeId, queryClient } = useImmunisationGraphQL();
+  const t = useTranslation('coldchain');
 
-  const mutationFn = async ({ name, programId }: DraftVaccineCourse) => {
-    return await api.insertVaccineCourse({
+  const mutationFn = async (input: DraftVaccineCourse) => {
+    const apiResult = await api.insertVaccineCourse({
       storeId,
       input: {
         id: FnUtils.generateUUID(),
-        name,
-        programId,
+        name: input.name,
+        programId: input.programId,
+        demographicIndicatorId: input.demographicIndicatorId,
+        coverageRate: input.coverageRate,
+        isActive: input.isActive,
+        wastageRate: input.wastageRate,
+        doses: input.doses,
+        vaccineItems:
+          input.vaccineCourseItems?.map(item =>
+            vaccineCourseParsers.toItemInput(item)
+          ) ?? [],
+        schedules:
+          input.vaccineCourseSchedules?.map(schedule =>
+            vaccineCourseParsers.toScheduleInput(schedule)
+          ) ?? [],
       },
     });
+
+    // will be empty if there's a generic error, such as permission denied
+    if (!isEmpty(apiResult)) {
+      const result = apiResult.centralServer.vaccineCourse.insertVaccineCourse;
+
+      if (result.__typename === 'VaccineCourseNode') {
+        return result;
+      }
+
+      let message: string;
+      switch (result.error.description) {
+        case InsertVaccineCourseError.RecordAlreadyExist:
+          message = t('error.database-error');
+          setErrorMessage(message);
+          throw new Error(
```

Comment:
I'm throwing error rather here than returning an error or setting error message because otherwise the default throw on line 205 would be called which wouldn't include error specifics